### PR TITLE
refactor!: remove uuid and rand dependencies where applicable

### PR DESCRIPTION
**Pull request by amrbashir**
ref: https://github.com/tauri-apps/tauri/issues/7756

### What kind of change does this PR introduce?

- [ ] Bugfix
- [ ] Feature
- [ ] Docs
- [ ] New Binding issue #___
- [ ] Code style update
- [X] Refactor
- [ ] Build-related changes
- [ ] Other, please describe:

### Does this PR introduce a breaking change?

- [X] Yes, and the changes were approved in issue #7756
- [ ] No

### Checklist
- [X] When resolving issues, they are referenced in the PR's title (e.g `fix: remove a typo, closes #___, #___`)
- [X] A change file is added if any packages will require a version bump due to this PR per [the instructions in the readme](https://github.com/tauri-apps/tauri/blob/dev/.changes/readme.md).
- [X] I have added a convincing reason for adding this feature, if necessary

### CODE REVIEW

**Comment by amrbashir** [comment]
There is a couple of places that still use `rnd` and `uuid` crates:
1. isolation pattern still uses `uuid`, not sure if we can replace it with a simple counter, cc @chippers
2. csp nonce and asset protocol range requests -> migrated to the lighter `getrandom` crate in [d398300](https://github.com/tauri-apps/tauri/pull/7939/commits/d398300ef0e0354ce18a48663e9e5f7fc9f19bd3)

**Comment by chippers** [comment]
> There is a couple of places that still use `rnd` and `uuid` crates:
> 
> 1. isolation pattern still uses `uuid`, not sure if we can replace it with a simple counter, cc @chippers
> 2. csp nonce and asset protocol range requests -> migrated to the lighter `getrandom` crate in [d398300](https://github.com/tauri-apps/tauri/pull/7939/commits/d398300ef0e0354ce18a48663e9e5f7fc9f19bd3)

it's used so that it's unpredictable and because the output chars are always valid as a schema suffix. using `getrandom` to generate 128 bits and using lowercase hex as the output should have the same effect here.

i've noticed some code that used uuid is replaced with smaller sizes of things, so take care that it's not problematic there. uuid v4 are often used as trusted random unique numbers because they are 128 bytes (well 122 since some are fixed) that you don't need to check for collisions for in most applications because it's so incredibly unlikely to get another of the same. u32s are much more likely to collide

some places they have been replaced with constants (like the event listener object and function names) which was seen as undesirable before to discourage people from using it manually and going through the proper api instead.

**Comment by amrbashir** [comment]
> it's used so that it's unpredictable and because the output chars are always valid as a schema suffix. using getrandom to generate 128 bits and using lowercase hex as the output should have the same effect here

I will leave that one using uuid for now, at least it is only active when using isolation pattern and the main goal for this PR was to remove it from public API.

> i've noticed some code that used uuid is replaced with smaller sizes of things, so take care that it's not problematic there. uuid v4 are often used as trusted random unique numbers because they are 128 bytes (well 122 since some are fixed) that you don't need to check for collisions for in most applications because it's so incredibly unlikely to get another of the same. u32s are much more likely to collide

the u32s are only incrementing so it shouldn't collide with itself, there maybe other ids that could have the same value but they are used in other places.

> some places they have been replaced with constants (like the event listener object and function names) which was seen as undesirable before to discourage people from using it manually and going through the proper api instead.

Yeah, I wasn't aware of the history behind that, I could use `getrandom` in this case but may I ask why it was discouraged? I mean we don't ever provide support for those who uses it directly and we can break it without any notice.

**Comment by chippers** [comment]
> Yeah, I wasn't aware of the history behind that, I could use `getrandom` in this case but may I ask why it was discouraged? I mean we don't ever provide support for those who uses it directly and we can break it without any notice.

just to avoid the typical "spacebar heating" problem if the behavior does change - and we should explicitly mark somewhere that it is unstable

**Comment by amrbashir** [comment]
could change the const from
```rs
function_name: "_listeners_function_id_",
object_name: "_listeners_object_id_",
```
to
```rs
function_name: "_internal_listeners_function_id_",
object_name: "_internal_listeners_object_id_",
```

File: .changes/tauri-runtime-uuid.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+'tauri-runtime': 'patch:breaking'
+---
+
+Added `WnidowEventId` type and Changed `Dispatch::on_window_event` return type from `Uuid` to `WindowEventId`
```
Comment (lucasfernog) [inline .changes/tauri-runtime-uuid.md]:
```suggestion
Added `WindowEventId` type and Changed `Dispatch::on_window_event` return type from `Uuid` to `WindowEventId`
```

File: .changes/tauri-uuid-rand.md [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+
+- Removed `EventHandler` type.
+- Added `EventId` type
+- Changed `Manager::listen_global` and `Window::listen` to return the new `EventId` type instead of `EventHandler`.
+- Removed the return type of `Manager::once_global` and `Window::once`
+- Changed `Manager::unlisten` and `Window::unlisten` to take he new `EventId` type.
+- Added `tauri::scope::ScopeEventId`
+- Changed `FsScope::listen` to return the new `ScopeEventId` instead of `Uuid`.
+- Added `FsScope::unlisten`
+-
```
Comment (lucasfernog) [inline .changes/tauri-uuid-rand.md]:
```suggestion

```

File: core/tauri/src/window.rs [hunk 3]
```diff
@@ -2228,15 +2228,15 @@ impl<R: Runtime> Window<R> {
...
     handler: CallbackFn,
-  ) -> crate::Result<usize> {
-    let event_id = rand::random();
+  ) -> crate::Result<EventId> {
+    let event_id = self.manager.listeners().next_event_id();
 
     self.eval(&crate::event::listen_js(
-      self.manager().event_listeners_object_name(),
-      format!("'{}'", event),
+      self.manager().listeners().function_name(),
```
Comment (lucasfernog) [inline core/tauri/src/window.rs]: this one was broken, changed listeners_object_name to function_name messed up the evaluated JS. I'm pushing a fix

**Review by lucasfernog — DISMISSED** [review]

**Review by lucasfernog — APPROVED** [review]

**Pull request merged**